### PR TITLE
When fetching Events, ensure that we try to fetch the Venue.

### PR DESCRIPTION
**Pull request by liuche**
Since our range for Event searching is greater than the range for Venues, we should ensure that we make a best-effort to fetch the venue (since we fetch it anyway).

### CODE REVIEW

**Comment by liuche** [comment]
[As a user I want my app to be in sync with the events stored in the firebase](https://trello.com/c/5PYxdd1V/212-as-a-user-i-want-my-app-to-be-in-sync-with-the-events-stored-in-the-firebase)

File: app/events.py [hunk 1]
```diff
@@ -101,9 +101,10 @@ def getGcalEventObj(event):
             if mapping:
                 placeName = mapping['name']
                 location = mapping['location']
-                yelpId = search._guessYelpId(placeName, location['lat'], location['lng'])
-                if (yelpId):
-                    eventObj = representation.eventRecord(yelpId, location['lat'], location['lng'], summary, event['start']['dateTime'], event['end']['dateTime'], event['htmlLink'])
+                yelpBiz = search._guessYelpBiz(placeName, location['lat'], location['lng'])
+                if (yelpBiz):
```
Comment (jhugman) [inline app/events.py:107]: Do we check if the venue is already in firebase? I'm a bit concerned we'll be burning up yelp search queries on venues we know about already.

File: app/events.py [hunk 2]
```diff
@@ -127,9 +128,10 @@ def fetchEventsFromLocation(latlong, maxResults, radius=5, dateRange="Today"):
 def getEventfulEventObj(event):
     locLat = event['latitude']
     locLng = event['longitude']
-    yelpId = search._guessYelpId(event['venue_name'], locLat, locLng)
-    if yelpId:
-        eventObj = representation.eventRecord(yelpId, locLat, locLng, event['title'], event['start_time'], event['stop_time'], event['url'])
+    yelpBiz = search._guessYelpBiz(event['venue_name'], locLat, locLng)
+    if yelpBiz:
+        researchVenue(yelpBiz)
```
Comment (jhugman) [inline app/events.py:135]: Same here. `_guessYelpBiz` burns an API query.

**Review by jhugman — COMMENTED** [review]
> Since our range for Event searching is greater than the range for Venues, we should ensure that we make a best-effort to fetch the venue (since we fetch it anyway).

I'm not sure I understand this point here. Is this the server searching for different sized geocircles? and if so, why? I understand why they'd be different on the client side.

Secondly, I'm concerned about burning yelp queries.

We should able to search for venues by lat/lon using firebase rather than immediately shelling out to yelp.

`request_handler.findSearchRecord` has most of that code to do that.

As counter points – 
1. we have 25k yelp queries per day.
2. `researchVenues` does aggressively cache everything. 
3. How often are we crawling these events, and how many are there?

**Comment by jhugman** [comment]
Merged now, but filed a follow-up #55.

**Pull request merged**